### PR TITLE
Deploy: t2 push will deploy a shell script containing the correct name corresponding to the deployed project's entry point. Fixes gh-460

### PR DESCRIPTION
**Pull request by rwaldron**

### CODE REVIEW

**Comment by HipsterBrown** [comment]
Looks good to me! :+1:

File: lib/tessel/deploy.js [hunk 1]
```diff
@@ -602,25 +605,23 @@ actions.writeToFile = function(t) {
...
 
-        // Read the contents of the node start script
-        fs.readFile(NODE_PUSH_SCRIPT, function(err, data) {
-          if (err) {
-            return reject(err);
-          }
-          // Write the Node start script
-          remoteProcess.stdin.end(data);
-        });
+        var shellScript = tags.stripIndent`
```
Comment (johnnyman727) [inline lib/tessel/deploy.js]: Cool lib! I hadn't seen that before.

**Comment by johnnyman727** [comment]
Code looks good besides Grunt's one peeve. Will do smoke test on my device right now.

**Comment by johnnyman727** [comment]
Mine isn't quite working:
```
INFO Looking for your Tessel...
INFO Connected to Frank.
INFO Writing project to Flash on Frank (3.072 kB)...
INFO Deployed.
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: Error upon remote process exit: 1
    at USBProcess.<anonymous> (/Users/Jon/Work/technical/t2-cli/lib/usb/usb_process.js:56:26)
    at USBProcess.g (events.js:260:16)
    at emitOne (events.js:77:13)
    at USBProcess.emit (events.js:169:7)
    at Parser.<anonymous> (/Users/Jon/Work/technical/t2-cli/lib/usb/usb_daemon.js:127:14)
    at emitOne (events.js:77:13)
    at Parser.emit (events.js:169:7)
    at Parser._write (/Users/Jon/Work/technical/t2-cli/node_modules/usb-daemon-parser/index.js:152:16)
    at doWrite (_stream_writable.js:292:12)
    at writeOrBuffer (_stream_writable.js:278:5)
    at Parser.Writable.write (_stream_writable.js:207:11)
    at ondata (_stream_readable.js:528:20)
    at emitOne (events.js:77:13)
    at emit (events.js:169:7)
    at readableAddChunk (_stream_readable.js:146:16)
    at Readable.push (_stream_readable.js:110:10)
```
Investigating now...

**Comment by johnnyman727** [comment]
But it does work over LAN.

**Comment by rwaldron** [comment]
What's the problem with USB? Is that happening when there is no LAN connection?

**Comment by rwaldron** [comment]
```
$ t2 run index.js --usb
INFO Looking for your Tessel...
INFO Connected to bishop.
INFO Generating slim build.
INFO Writing project to RAM on bishop (2.048 kB)...
INFO Deployed.
INFO Running bundled project...
I'm blinking! (Press CTRL + C to stop)
```

**Comment by johnnyman727** [comment]
It looks like it must be a timing issue. If I print `stderr` on the `setExecutablePermissions` process, I get:
```
err chmod: /app/start: No such file or directory
```

I think the file isn't actually finished writing when the permissions are set.

In any case, I think this is unrelated to your change and the issue already exists: https://github.com/tessel/t2-cli/issues/415

We should just merge if it's :+1: with Travis.

**Comment by rwaldron** [comment]
Ok, travis is green, will land.

**Pull request merged**